src/Microsoft.AspNet.Authentication/AuthenticationOptions.cs

```diff
@@ -42,5 +43,11 @@ public string AuthenticationScheme
         /// Additional information about the authentication type which is made available to the application.
         /// </summary>
         public AuthenticationDescription Description { get; set; } = new AuthenticationDescription();
+
+        /// <summary>
+        /// The request path within the application's base path where the user-agent will be returned
+        /// when a failure occurs.
+        /// </summary>
+        public PathString ErrorHandlerPath { get; set; }
```

Comment:
Perhaps RedirectingAuthenticationMiddleware as the name of the new shared base class for OAuth/OIDC/Twitter